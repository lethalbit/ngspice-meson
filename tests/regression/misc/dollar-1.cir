test $var expansion

v1 1 0 dc = 0

.control

set foo = ""
echo "TEST:" ">{$foo}< should be ><"

set foo = ( 1 2 3 )
set bar = 2
echo "TEST:" ">$foo[0]< should be >1<"
echo "TEST:" ">$foo[$bar]< should be >2<"

set foo = ( $foo baz  bar )
echo "TEST:" $foo
echo "TEST:" "should be >1 2 3 baz bar<"
echo "TEST:" ">$foo[0]< should be >1<"

set aux = ( )
echo "TEST:" $aux "end"

set foo = ( $aux mi au )
set bar = ( mi $aux au )
set baz = ( mi au $aux )

echo "TEST:" $foo
echo "TEST:" "should be >mi au<"
echo "TEST:" ">$foo[0]< should be >mi<"

echo "TEST:" $bar
echo "TEST:" "should be >mi au<"
echo "TEST:" ">$bar[0]< should be >mi<"

echo "TEST:" $baz
echo "TEST:" "should be >mi au<"
echo "TEST:" ">$baz[0]< should be >mi<"

quit 0

.endc

.end
