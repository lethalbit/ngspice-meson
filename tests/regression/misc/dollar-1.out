
Circuit: test for regression of already fixed bugs

TEST: >< should be ><
TEST: >1< should be >1<
TEST: >2< should be >2<
TEST: 1 2 3 baz bar
TEST: should be >1 2 3 baz bar<
TEST: >1< should be >1<
TEST: end
TEST: mi au
TEST: should be >mi au<
TEST: >mi< should be >mi<
TEST: mi au
TEST: should be >mi au<
TEST: >mi< should be >mi<
TEST: mi au
TEST: should be >mi au<
TEST: >mi< should be >mi<

Doing analysis at TEMP = 27.000000 and TNOM = 27.000000


No. of Data Rows : 1
